### DISCUSSION TITLE
Welcome to RaspberryPi Discussions!

### DISCUSSION THREAD

**Discussion by DaveStoneHub** [ATGNI/RaspberryPi#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by 57gmc** [comment]
Hi Dave, I'm trying to debug a Hello World console app following the directions in your video. I can't get the app to run on my RPi 4 B. There is something not right in the launch.json remote configuration. The app get published to the Pi just fine, but it fails to run. I get an error that says:
launch: program " does not exist.
I can ssh to the project folder on the pi and run; dotnet ConsoleTest.dll and it prints "Hello World!". So I know the code works. It just can't find it. I've tried using explicit paths but it still can't find it.
This is the only output in the Debug Console window.: Starting: "ssh" pi@raspberrypi "/home/pi/vsdbg/vsdbg --interpreter=vscode"

**Reply by 57gmc** [reply]
I had to change
"program": "dotnet",
to
"program": "~/.dotnet/dotnet",

**Comment by DaveStoneHub** [comment]
Thanks for the update.. Good detective work.. Mine still works, so I think it’s a case of how it finds the Dotnet executable. This is the result of my Pi:

***@***.***

This is what I have in my ~/.dotnet directory….

***@***.***

Could you send me your results from “whereis” and also list your .dotnet directory. I’ll update the github instructions for those that might experience a similar problem..

Thanks for reaching out.

Dave

**Reply by 57gmc** [reply]
The first time, I had started from Microsoft's directions for deploying IoT. It installed dotnet to $Home/.dotnet. I got VS Code to run, but the debugger wouldn't work. So, I thought if I started from scratch using your directions it might help. Your dotnet deployment installed dotnet to /opt/dotnet and I'm unable to get the os to find it because it's up a level from $Home. On Microsoft's page for [deploying dotnet to a Pi](https://learn.microsoft.com/en-us/dotnet/iot/deployment), they show how to set the path variable, but I can't get that to work for when dotnet is installed under /opt.

**Reply by 57gmc** [reply]
I finally got it working. I used Microsoft's instructions in the previous link to install dotnet and vsdbg and the rest of your instructions for setting up VS Code, file share, etc. Since I have the 4B and the debugger installs the arm64 version, I decided to install Raspbian 64 bit. Maybe the reason I couldn't get the debugger to work before was a version conflict. The only code of your that I had to change was the launch.json file. Below is my file as a txt.
[launch.txt](https://github.com/ATGNI/RaspberryPi/files/11070312/launch.txt)
